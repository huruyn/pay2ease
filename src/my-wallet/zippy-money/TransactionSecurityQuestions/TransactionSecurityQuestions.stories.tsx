import { ComponentStory, ComponentMeta } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import TransactionSecurityQuestions from './TransactionSecurityQuestions';
import MAIN_INFO from '../../../stories/MainInfo';

export default {
  title: 'my-wallet/zippy-money/TransactionSecurityQuestions',
  component: TransactionSecurityQuestions,
} as ComponentMeta<typeof TransactionSecurityQuestions>;

const Template: ComponentStory<typeof TransactionSecurityQuestions> = () => (
  <BrowserRouter>
    <TransactionSecurityQuestions
      mainInfo={MAIN_INFO}
      setMainInfo={jest.fn()}
      isProcessing={false}
      handleTriggerTransaction={jest.fn()}
    />
  </BrowserRouter>
);

export const Default = Template.bind({});
Default.args = {};
Default.parameters = {};

export const APIError = Template.bind({});
APIError.args = {};
APIError.parameters = {};
