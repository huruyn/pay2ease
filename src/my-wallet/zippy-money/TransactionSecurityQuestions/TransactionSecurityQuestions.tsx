import {
  Button,
  FormControl,
} from 'react-bootstrap';
import { useState } from 'react';
import PageContainer from '../../../common/PageContainer';
import { TransactionInterface } from '../../../constants/interface/TransactionInterface';

export default function TransactionSecurityQuestions({
  mainInfo,
  setMainInfo,
  handleTriggerTransaction,
  isProcessing,
}: Pick<TransactionInterface, 'mainInfo' | 'setMainInfo' | 'handleTriggerTransaction' | 'isProcessing'>) {
  const [confirmAnswer, setConfirmAnswer] = useState<string>('');
  const handleSubmit = () => {
    if (mainInfo.securityQuestion.length > 40 || mainInfo.securityAnswer !== confirmAnswer) {
      return;
    }
    handleTriggerTransaction();
  };

  return (
    <>
      <PageContainer
        title="Personal Account"
        subTitle="Made Fun With Zippy!"
        backdropImage="backdrop-image-2"
      >
        <div className="title">
          Security Question
        </div>
        <div className="security-question">
          <FormControl
            placeholder="Create a security question (Up to 40 Char.)"
            value={mainInfo.securityQuestion}
            onChange={(evt) => setMainInfo({ ...mainInfo, securityQuestion: evt.target.value })}
          />
          <FormControl
            placeholder="Security Answer (No Space)"
            type={mainInfo.showAnswer ? 'text' : 'password'}
            value={mainInfo.securityAnswer}
            onChange={(evt) => setMainInfo({ ...mainInfo, securityAnswer: evt.target.value })}
          />
          <FormControl
            placeholder="Re-enter Security Answer"
            type={mainInfo.showAnswer ? 'text' : 'password'}
            value={confirmAnswer}
            onChange={(evt) => setConfirmAnswer(evt.target.value)}
          />
        </div>
        <div className="action">
          <Button
            className="zippy-btn"
            onClick={handleSubmit}
            disabled={isProcessing}
          >
            {isProcessing && <div className="loading spinner-border" role="status" />}
            Zipp It
          </Button>
        </div>
      </PageContainer>
    </>
  );
}
