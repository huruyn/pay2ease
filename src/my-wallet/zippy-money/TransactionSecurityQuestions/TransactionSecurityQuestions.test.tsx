import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import TransactionSecurityQuestions from './TransactionSecurityQuestions';
import MAIN_INFO from '../../../stories/MainInfo';

const ReactTestRenderer = require('react-test-renderer');

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useParams: () => ({
    stepId: 'details',
  }),
  useNavigate: () => mockedUsedNavigate,
}));

describe('TransactionSecurityQuestions Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <BrowserRouter>
        <TransactionSecurityQuestions
          mainInfo={MAIN_INFO}
          setMainInfo={jest.fn()}
          isProcessing={false}
          handleTriggerTransaction={jest.fn()}
        />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
