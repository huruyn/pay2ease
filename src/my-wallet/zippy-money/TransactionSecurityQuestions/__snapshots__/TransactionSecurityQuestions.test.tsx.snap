// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionSecurityQuestions Component matches the snapshot 1`] = `
<div
  className=""
>
  <div
    className="zippy-cash-container-wrap"
  >
    <div
      className="backdrop backdrop-image-2"
    />
    <div
      className="zippy-cash-container"
    >
      <div
        className="zippy-cash-container-header"
      >
        <h1
          className="title"
        >
          Personal Account
        </h1>
        <h2
          className="catch"
        >
          Made Fun With Zippy!
        </h2>
      </div>
      <div
        className="zippy-cash-container-body"
      >
        <div
          className="title"
        >
          Security Question
        </div>
        <div
          className="security-question"
        >
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="Create a security question (Up to 40 Char.)"
            value="security Question"
          />
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="Security Answer (No Space)"
            type="text"
            value="security Answer"
          />
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="Re-enter Security Answer"
            type="text"
            value=""
          />
        </div>
        <div
          className="action"
        >
          <button
            className="zippy-btn btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Zipp It
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
